src/Controls/tests/DeviceTests/Elements/Layout/LayoutTests.cs

```diff
@@ -92,5 +90,64 @@ public async Task CascadeInputTransparentAppliesOnInit(bool inputTransparent, bo
 
 			Assert.Equal(expected, child.InputTransparent);
 		}
+
+		[Theory(
+#if IOS
+			Skip = "Not able to debug iOS right now"
+#endif
+		)]
+		[InlineData(typeof(Grid), LayoutAlignment.Center)]
+		[InlineData(typeof(Grid), LayoutAlignment.Start)]
+		[InlineData(typeof(Grid), LayoutAlignment.End)]
+		[InlineData(typeof(VerticalStackLayout), LayoutAlignment.Center)]
+		[InlineData(typeof(VerticalStackLayout), LayoutAlignment.Start)]
+		[InlineData(typeof(VerticalStackLayout), LayoutAlignment.End)]
+		public async Task UpdatingLayoutOptionsTriggersParentToRepositionControl(Type layoutType, LayoutAlignment layoutAlignment)
+		{
+			var layoutOptions = new LayoutOptions(layoutAlignment, false);
+
+			// create a layout with the values all set before creating the handler
+			CreateLayout(layoutType, out var initialLayout, out var initialLabel);
+			initialLabel.HorizontalOptions = layoutOptions;
+
+			// create a layout that will update once attached
+			CreateLayout(layoutType, out var updatingLayout, out var updatingLabel);
+
+			await InvokeOnMainThreadAsync(async () =>
+			{
+				_ = CreateHandler<LabelHandler>(initialLabel);
+				var initialHandler = CreateHandler<LayoutHandler>(initialLayout);
+				var initialBitmap = await initialHandler.PlatformView.ToBitmap();
+
+				_ = CreateHandler<LabelHandler>(updatingLabel);
+				var updatingHandler = CreateHandler<LayoutHandler>(updatingLayout);
+				var updatingBitmap = await updatingHandler.PlatformView.AttachAndRun(() =>
+				{
+					updatingLabel.HorizontalOptions = layoutOptions;
+
+					return updatingHandler.PlatformView.ToBitmap();
+				});
+
+				await initialBitmap.AssertEqual(updatingBitmap);
+			});
```

Comment:
Bitmap comparison tests are great. Might need a fuzzer at some point, but for now if the test is running both UIs for comparison, then is should be exactly the same. Also works great on Android. Not so much iOS yet #6832